go/src/lib.rs

```diff
@@ -83,30 +83,30 @@ fn create_client_internal(
     let client = runtime
         .block_on(GlideClient::new(ConnectionRequest::from(request)))
         .map_err(|err| err.to_string())?;
-    Ok(Client {
+    Ok(ClientContainer {
         client,
         success_callback,
         failure_callback,
         runtime,
     })
 }
 
-/// Creates a new client with the given configuration.
+/// Creates a new `ClientContainer` with a new `GlideClient` configured using a Protobuf `ConnectionRequest`.
 ///
-/// The returned `ConnectionResponse` should be manually freed by calling `free_connection_response`, otherwise a memory leak will occur. It should be freed whether or not an error occurs.
+/// The returned `ConnectionResponse` will only be freed by calling `free_connection_response`.
 ///
 /// `connection_request_bytes` is an array of bytes that will be parsed into a Protobuf `ConnectionRequest` object.
 /// `connection_request_len` is the number of bytes in `connection_request_bytes`.
-/// `success_callback` is the callback that will be called in the case that the Redis command succeeds.
-/// `failure_callback` is the callback that will be called in the case that the Redis command fails.
+/// `success_callback` is the callback that will be called when a Redis command succeeds.
+/// `failure_callback` is the callback that will be called when a Redis command fails.
 ///
 /// # Safety
 ///
 /// * `connection_request_bytes` must point to `connection_request_len` consecutive properly initialized bytes. It must be a well-formed Protobuf `ConnectionRequest` object. The array must be allocated by the caller and subsequently freed by the caller after this function returns.
 /// * `connection_request_len` must not be greater than the length of the connection request bytes array. It must also not be greater than the max value of a signed pointer-sized integer.
-/// * The `conn_ptr` pointer in the returned `ConnectionResponse` must live until it is passed into `close_client`.
+/// * The `conn_ptr` pointer in the returned `ConnectionResponse` must live while the client is open/active and must be explicitly freed by calling `close_client`.
 /// * The `connection_error_message` pointer in the returned `ConnectionResponse` must live until the returned `ConnectionResponse` pointer is passed to `free_connection_response`.
-/// * Both the `success_callback` and `failure_callback` function pointers need to live until the client is closed via `close_client` since they are used when issuing Redis commands. In the case of an error, the function pointers live until the client is explicitly closed.
+/// * Both the `success_callback` and `failure_callback` function pointers need to live while the client is open/active. The caller is responsible for deallocating both callbacks.
```

Comment:
here you're using "deallocating", in line 96 you use "freed", and in line 37 you use "destroyed". unless you're making some fine distinction here (if you are, I can't see it), pick a single term and stick with it.